pkg/resource/plugin/rpc.go

```diff
@@ -340,6 +351,13 @@ func UnmarshalPropertyValue(v *structpb.Value, opts MarshalOptions) (*resource.P
 			}
 			s := resource.MakeSecret(value)
 			return &s, nil
+		case resource.ResourceSig:
+			urn, ok := obj["urn"]
+			if !ok {
+				return nil, errors.New("malformed RPC resource: missing urn")
```

Comment:
```suggestion
				return nil, errors.New("malformed RPC resource: missing urn for ResourceSig")
```

maybe?  otherwise the message just seems a little too vague.